cdap-app-fabric/src/main/java/co/cask/cdap/templates/AdapterDefinition.java

```diff
@@ -145,6 +152,16 @@ public JsonElement getConfig() {
     return config;
   }
 
+  @Override
+  public <T> T getConfig(Type configType) {
+    return GSON.fromJson(config, configType);
+  }
+
+  @Override
+  public String getConfigString() {
+    return config.toString();
```

Comment:
This is the one that extends ``PluginConfig``, right? Its ``toString()`` returns a JSON?